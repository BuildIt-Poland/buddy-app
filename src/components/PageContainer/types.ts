import { ContainerProps } from '@material-ui/core/Container';

export interface PageContainerProps extends ContainerProps {
  loading?: boolean;
}
