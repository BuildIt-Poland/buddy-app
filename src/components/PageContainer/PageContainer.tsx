import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';
import BackgroundShape from 'components/BackgroundShape';
import { PageContainerProps } from './types';

const useStyles = makeStyles(theme => ({
  container: {
    [theme.breakpoints.down('sm')]: {
      paddingTop: theme.spacing(2),
    },
    [theme.breakpoints.up('sm')]: {
      paddingTop: theme.spacing(3),
    },
  },
}));

const PageContainer: React.FC<PageContainerProps> = props => {
  const classes = useStyles();
  const { children, backGroundShape, ...args } = props;
  return (
    <Container component={'main'} className={classes.container} {...args}>
      {children}
      {backGroundShape && <BackgroundShape />}
    </Container>
  );
};

export default PageContainer;
