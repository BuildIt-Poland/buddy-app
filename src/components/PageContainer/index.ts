import PageContainers from './PageContainer';

export default PageContainers;
