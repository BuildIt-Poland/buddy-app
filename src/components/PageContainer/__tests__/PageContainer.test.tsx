import React from 'react';
import { create } from 'react-test-renderer';
import PageContainer from '../PageContainer';

jest.mock('@material-ui/core/Container', () => 'Container');
jest.mock('@material-ui/core/LinearProgress', () => 'LinearProgress');
jest.mock('components/BackgroundShape', () => 'BackgroundShape');

describe('Component - PageContainer', () => {
  describe('when is loading', () => {
    it('renders correctly', async () => {
      const component = create(<PageContainer loading>hello world</PageContainer>);

      expect(component.toJSON()).toMatchSnapshot();
    });
  });

  describe('when is not loading', () => {
    it('renders correctly', async () => {
      const component = create(<PageContainer>hello world</PageContainer>);

      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
