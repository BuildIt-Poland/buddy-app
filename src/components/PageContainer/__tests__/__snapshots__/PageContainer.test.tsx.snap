// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - PageContainer when is loading renders correctly 1`] = `
<Container
  className="makeStyles-container-1"
  component="main"
  loading={true}
>
  hello world
</Container>
`;

exports[`Component - PageContainer when is not loading renders correctly 1`] = `
<Container
  className="makeStyles-container-1"
  component="main"
>
  hello world
</Container>
`;
